src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,192 @@
+package com.minecolonies.coremod.util;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.entity.ai.mobs.EntityArcherBarbarian;
+import com.minecolonies.coremod.entity.ai.mobs.EntityBarbarian;
+import com.minecolonies.coremod.entity.ai.mobs.EntityChiefBarbarian;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+
+import java.util.stream.IntStream;
+
+/**
+ * Utils for The Barbarian Raid Event
+ */
+public class BarbarianUtils
+{
+    /**
+     * ResourceLocations for barbarians
+     */
+    /* default */ public static ResourceLocation barbarian = EntityList.getKey(EntityBarbarian.class);
+    /* default */ public static ResourceLocation archer    = EntityList.getKey(EntityArcherBarbarian.class);
+    /* default */ public static ResourceLocation chief     = EntityList.getKey(EntityChiefBarbarian.class);
+
+    public static void doRaid(World world, final int level, Colony colony)
+    {
+        eventRaid(world, level, colony);
+    }
+
+    /**
+     * Commences a Raid event in a specific colony, with a specific RaidLevel
+     *
+     * @param raidingWorld
+     * @param level
+     * @param colony
+     */
+
+    private static void eventRaid(World raidingWorld, final int level, Colony colony)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "eventRaid" is 20 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)